src/common/image.c

```diff
@@ -1091,51 +1091,17 @@ void dt_image_remove(const int32_t imgid)
   if(darktable.gui && darktable.gui->expanded_group_id == old_group_id)
     darktable.gui->expanded_group_id = new_group_id;
 
+   // explicitly enable foreign keys
```

Comment:
likewise

Reply:
I would agree completely with you if foreign keys had not been disabled by accident in legacy_presets.h until db version 33; they were therefore also (most probably, but I had not tested) disabled for data.locations which has had one for quite some time. 
Will nevertheless cut those lines as requested, now that I'll watch closely changes to db structure :rofl:

Reply:
Thanks! Can you squash the removals of pragma in corresponding commits ? If you don't know how to do that I'll do it. I'd like to avoid merging a commit and a partial revert, to me this should not be part of history expect as discussion in this PR. TIA.